File: CPP/Aksesoris.cpp
#include <iostream>
#include <string>
#include <iomanip>
using namespace std;
#include "PetShop.cpp" // Mengimpor file header PetShop.cpp

// Kelas Aksesoris yang merupakan turunan dari kelas PetShop
class Aksesoris : public PetShop {
private:
    string jenis; // Jenis aksesoris
    string bahan; // Bahan aksesoris
    string warna; // Warna aksesoris

public:
    // Constructor default
    Aksesoris() : PetShop(), jenis(""), bahan(""), warna("") {}

    // Constructor dengan parameter untuk menginisialisasi atribut, termasuk dari kelas PetShop
    Aksesoris(string id, string nama_produk, int harga_produk, int stok_produk, string jenis, string bahan, string warna)
        : PetShop(id, nama_produk, harga_produk, stok_produk) {
        this->jenis = jenis;   // Mengatur jenis aksesoris
        this->bahan = bahan;   // Mengatur bahan aksesoris
        this->warna = warna;   // Mengatur warna aksesoris
    }

    // Setter untuk jenis aksesoris
    void setJenis(string jenis) { 
        this->jenis = jenis; 
    }

    // Getter untuk mendapatkan jenis aksesoris
    string getJenis() { 
        return jenis; 
    }

    // Getter untuk mendapatkan bahan aksesoris
    string getBahan() { 
        return bahan; 
    }

    // Setter untuk bahan aksesoris
    void setBahan(string bahan) { 
        this->bahan = bahan; 
    }

    // Setter untuk warna aksesoris
    void setWarna(string warna) { 
        this->warna = warna; 
    }

    // Getter untuk mendapatkan warna aksesoris
    string getWarna() { 
        return warna; 
    }

    // Method display untuk menampilkan informasi aksesoris dengan tambahan atribut baru
    void display(int id_max, int nama_max, int harga_max, int stok_max, int jenis_max, int bahan_max, int warna_max) {
        // Memanggil method display dari kelas induk (PetShop) untuk menampilkan atribut dasar
        PetShop::display(id_max, nama_max, harga_max, stok_max);
        
        // Menampilkan atribut tambahan milik Aksesoris dengan format yang rapi
        cout << jenis << string(jenis_max - jenis.size(), ' ')
             << bahan << string(bahan_max - bahan.size(), ' ')
             << warna << string(warna_max - warna.size(), ' ');
    }
};

File: PHP/Aksesoris.php
<?php

// Mengimpor file PetShop.php agar dapat menggunakan kelas PetShop sebagai parent class
include ("PetShop.php");

// Mendefinisikan kelas Aksesoris yang merupakan turunan dari kelas PetShop
class Aksesoris extends PetShop {
    // Mendeklarasikan properti tambahan yang dimiliki oleh Aksesoris
    private $jenis;
    private $bahan;
    private $warna;

    // Konstruktor default untuk menginisialisasi properti
    public function __construct($id = "", $nama_produk = "", $harga_produk = 0, $stok_produk = 0, $foto = "",
                               $jenis = "", $bahan = "", $warna = "") {
        // Memanggil konstruktor dari kelas induk (PetShop) untuk mengatur properti yang diwarisi
        parent::__construct($id, $nama_produk, $harga_produk, $stok_produk, $foto);

        // Menginisialisasi properti khusus dari kelas Aksesoris
        $this->jenis = $jenis;
        $this->bahan = $bahan;
        $this->warna = $warna;
    }

    // Metode untuk mengatur jenis aksesoris
    public function setJenis($jenis) {
        $this->jenis = $jenis;
    }

    // Metode untuk mendapatkan jenis aksesoris
    public function getJenis() {
        return $this->jenis;
    }

    // Metode untuk mengatur bahan aksesoris
    public function setBahan($bahan) {
        $this->bahan = $bahan;
    }

    // Metode untuk mendapatkan bahan aksesoris
    public function getBahan() {
        return $this->bahan;
    }

    // Metode untuk mengatur warna aksesoris
    public function setWarna($warna) {
        $this->warna = $warna;
    }

    // Metode untuk mendapatkan warna aksesoris
    public function getWarna() {
        return $this->warna;
    }

    // Metode display() dari kelas induk untuk menampilkan informasi aksesoris
    public function display($id_max, $nama_max, $harga_max, $stok_max, $foto_max = null, $jenis_max = null, $bahan_max = null, $warna_max = null) {
        // Memanggil metode display() dari kelas induk untuk menampilkan informasi dasar produk
        parent::display($id_max, $nama_max, $harga_max, $stok_max, $foto_max);
        
        // Menampilkan informasi tambahan jika parameter maksimal untuk atribut tambahan diberikan
        if ($jenis_max !== null && $bahan_max !== null && $warna_max !== null) {
            echo $this->jenis . str_repeat(" ", $jenis_max - strlen($this->jenis))
                 . $this->bahan . str_repeat(" ", $bahan_max - strlen($this->bahan))
                 . $this->warna . str_repeat(" ", $warna_max - strlen($this->warna));
        }
    }    
}
?>
